test/customValidators_test.js

```diff
@@ -1076,6 +1077,273 @@ describe('validateOperationId()', function () {
   });
 });
 
+describe('validateMessageId()', function () {
+  const operations = ['subscribe', 'publish'];
+
+  it('should successfully validate messageId', async function () {
+    const inputString = `{
+      "asyncapi": "2.0.0",
+      "info": {
+        "version": "1.0.0"
+      },
+      "channels": {
+        "test/1": {
+          "publish": {
+            "message": {
+              "messageId": "test1"
+            }
+          }
+        },
+        "test/2": {
+          "subscribe": {
+            "message": {
+              "messageId": "test2"
+            }
+          }
+        }
+      }
+    }`;
+    const parsedInput = JSON.parse(inputString);
+
+    expect(
+      validateMessageId(parsedInput, inputString, input, operations)
+    ).to.equal(true);
+  });
+
+  it('should throw error that messageIds are duplicated and that they duplicate', async function () {
+    const inputString = `{
+      "asyncapi": "2.0.0",
+      "info": {
+        "version": "1.0.0"
+      },
+      "channels": {
+        "test/1": {
+          "publish": {
+            "message": {
+              "messageId": "test"
+            }
+          }
+        },
+        "test/2": {
+          "subscribe": {
+            "message": {
+              "messageId": "test"
+            }
+          }
+        },
+        "test/3": {
+          "subscribe": {
+            "message": {
+              "messageId": "test"
+            }
+          }
+        },
+        "test/4": {
+          "subscribe": {
+            "operationId": "test4"
+          }
+        }
+      }
+    }`;
+    const parsedInput = JSON.parse(inputString);
+
+    const expectedErrorObject = {
+      type: 'https://github.com/asyncapi/parser-js/validation-errors',
+      title: 'messageId must be unique across all the messages.',
+      parsedJSON: parsedInput,
+      validationErrors: [
+        {
+          title:
+            'test/2/subscribe/message/messageId is a duplicate of: test/1/publish/message/messageId',
```

Comment:
You should use tilde also in titles and full path

```suggestion
            '/channels/test~12/subscribe/message/messageId is a duplicate of: channels/test~11/publish/message/messageId',
```

etc. Please fix it in the `validateMessageId` function.

Reply:
We are using tilde for title, but it gets untidle in the `utils.groupValidationErrors` method.

https://github.com/asyncapi/parser-js/blob/d3dd11d885074c29ef5cde8956c5968351a54d7a/lib/utils.js#L259

Reply:
Ok, I forgot about it, please go ahead and don't worry about my comment. We should fix it in the another PR.